src/app.ts

```diff
@@ -13,14 +14,61 @@ interface TemperatureSummary {
   average: number
 }
 
+/* Object containing a dictionary of dictionaries */
+const sumarys: { [key: string]: { [item: string]: TemperatureSummary } } = {}
+
+/* Function that given the array of TemperatureReadings fills the sumarys object,
+  the first dictionaries with cities and the dictionary inside of the cities
+  with the dates containing the sumary of that day */
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (let read of readings) {
+    let exist = read.city in sumarys ? (read.time.toDateString() in sumarys[read.city]) ? true : false : false
+    if (!exist) {
+      const cityReadings = readings.filter(
+        (item) =>
+          item.city === read.city &&
+          item.time.toDateString() === read.time.toDateString(),
+      )
+      let cityFirst = cityReadings[0]
+      let cityLast = cityReadings[cityReadings.length - 1]
+      let cityHigh = cityReadings.reduce((a, b) =>
+        a.temperature > b.temperature ? a : b,
+      )
+      let cityLow = cityReadings.reduce((a, b) =>
+        a.temperature < b.temperature ? a : b,
+      )
```

Comment:
The high and the low firstly should be for the day. What is the high for today and hat is the low for today. You can also add in a high and low for the city for all the readings. Either way I don't think you need to use reduce. Reduce is just adding to the complexity of the function by adding extra loops to each loop. `Math.min` and `Math.max` between 2 numbers should be good enough